lua/java/api/profile_config.lua

```diff
@@ -0,0 +1,68 @@
+local log = require('java.utils.log')
+local async = require('java-core.utils.async').sync
+local get_error_handler = require('java.handlers.error')
+
+local config_path = vim.fn.stdpath('data') .. '/nvim-java-profiles.json'
+local current_project_path = vim.fn.expand('%:p:h')
+
+--- @class ProjectConfig
+--- @field profiles table
+local ProjectConfig = {}
+function ProjectConfig:new()
```

Comment:
Hi, could you replace the class constructor with penlight class. It's already in the java-core.utils.class. You can directly import from java-core

Here is a sample code.
https://github.com/nvim-java/nvim-java-core/blob/308c17c732eb29f5332b1f5e0e347206591f8660/lua/java-core/ls/clients/jdtls-client.lua?plain=1#L39-L43

More info
https://stevedonovan.github.io/Penlight/api/libraries/pl.class.html